### PR TITLE
Initial Tracing extraction.

### PR DESCRIPTION
**Pull request by alesj**
Introducing OpenTelemetry tracing option.

The new tracing module abstracts away the tracing usage -- either previous OpenTracing or new OpenTelemetry support. Instead of same tracing config code being applied over all test-client modules, we handle this in a single tracing module.
Different tracing implementations are triggered by setting implementation specific service name environment variable (see tracing's README for more details).
Each tracing implementation sets a global Tracer instance, which can then be used in supported libraries to provide new traces / spans.
Additionally to service name env var, other required env vars must be set as well -- see README.
Currently tracing implementations add support for Kafka consumer/producer and Kafka Streams tracing.

Similar implementation is used in Strimzi Operator and Strimzi Bridge,
so it might be useful to reconsider a better place for all this implementation code / artifact.

Signed-off-by: Ales Justin <ales.justin@gmail.com>

### CODE REVIEW

File: docker-images/build-images.sh [hunk 1]
```diff
@@ -7,6 +7,12 @@ MODE=${1:-"push"}
 DOCKER_TAG=${DOCKER_TAG:-"latest"}
 MVN_ARGS=${MVN_ARGS:-""}
 
+echo "Installing root pom"
+make mvn_build
+
+echo "Building Tracing dependency"
+make build --directory=tracing/
+
```
Comment (scholzj) [inline docker-images/build-images.sh:15]: I'm not sure I understand why is this needed. Can you elaborate on this a bit?

File: Makefile [hunk 2]
```diff
@@ -15,6 +16,9 @@ $(DOCKER_TARGETS): $(SUBDIRS)
 $(SUBDIRS):
 	$(MAKE) -C $@ $(MAKECMDGOALS)
 
+mvn_build:
```
Comment (scholzj) [inline Makefile:19]: Should this be somewhere in `Makefile.maven` instead of here and be called `java_install` or something like that?

File: http/http-producer/pom.xml [hunk 3]
```diff
@@ -32,12 +32,8 @@
             <version>${log4j.version}</version>
         </dependency>
         <dependency>
-            <groupId>io.jaegertracing</groupId>
-            <artifactId>jaeger-client</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>io.opentracing.contrib</groupId>
-            <artifactId>opentracing-kafka-client</artifactId>
```
Comment (scholzj) [inline http/http-producer/pom.xml:40]: What does removing of the OpenTracing stuff mean for backwards compatibility and testing the OpenTracing stuff?

File: kafka/admin/src/main/java/AdminConfiguration.java [hunk 4]
```diff
@@ -93,9 +93,9 @@ public static Properties createProperties(AdminConfiguration config) {
             }
         }
 
-        if ((config.getOauthAccessToken() != null)
-                || (config.getOauthTokenEndpointUri() != null && config.getOauthClientId() != null && config.getOauthRefreshToken() != null)
-                || (config.getOauthTokenEndpointUri() != null && config.getOauthClientId() != null && config.getOauthClientSecret() != null))    {
+        if (config.getOauthAccessToken() != null
+                || config.getOauthTokenEndpointUri() != null && config.getOauthClientId() != null && config.getOauthRefreshToken() != null
+                || config.getOauthTokenEndpointUri() != null && config.getOauthClientId() != null && config.getOauthClientSecret() != null)    {
```
Comment (scholzj) [inline kafka/admin/src/main/java/AdminConfiguration.java:98]: Can we keep the old code? I think it makes the code more readable.

File: kafka/consumer/src/main/java/ConsumerConfiguration.java [hunk 5]
```diff
@@ -102,9 +102,9 @@ public static Properties createProperties(ConsumerConfiguration config) {
             }
         }
 
-        if ((config.getOauthAccessToken() != null)
-            || (config.getOauthTokenEndpointUri() != null && config.getOauthClientId() != null && config.getOauthRefreshToken() != null)
-            || (config.getOauthTokenEndpointUri() != null && config.getOauthClientId() != null && config.getOauthClientSecret() != null))    {
+        if (config.getOauthAccessToken() != null
+            || config.getOauthTokenEndpointUri() != null && config.getOauthClientId() != null && config.getOauthRefreshToken() != null
```
Comment (scholzj) [inline kafka/consumer/src/main/java/ConsumerConfiguration.java:106]: Same as above ...

**Review by scholzj — COMMENTED** [review]
There are some things which I didn't really understood - what was their motivation etc. I didn't understand some of the changes to the build system. And I didn't really understand why is the new module needed. Maybe you can explain this in the PR description to make it clear what and why is changed and how is it expected to work? As a sort fo mini-design-doc? Since it is not a simple bugfix on few lines, understanding it is important for reviewing the PR.

File: tracing/src/main/java/io/strimzi/test/tracing/TracingUtil.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.apache.kafka.streams.KafkaClientSupplier;
+
+/**
+ * Tracing initialization
+ */
+public class TracingUtil {
+    private static final Logger log = Logger.getLogger(TracingUtil.class.getName());
```
Comment (Frawless) [inline tracing/src/main/java/io/strimzi/test/tracing/TracingUtil.java:18]: why you don't use log4j as we do in the whole project?

**Review by Frawless — COMMENTED** [review]
I agree with Jakub, better PR description and an example of usage in README would be very helpful

File: kafka/admin/src/main/java/AdminConfiguration.java (hunk 4 above)
Comment (im-konge) [reply kafka/admin/src/main/java/AdminConfiguration.java:98]: Agree with Jakub

File: kafka/producer/src/main/java/Main.java [hunk 7]
```diff
@@ -80,7 +71,7 @@ public static void main(String[] args) throws InterruptedException {
                 // Increment number of sent messages for non blocking producer
                 numSent.incrementAndGet();
             }
-            if (transactionalProducer && ((i + 1) % msgPerTx == 0)) {
+            if (transactionalProducer && (i + 1) % msgPerTx == 0) {
```
Comment (im-konge) [inline kafka/producer/src/main/java/Main.java:74]: Same as above, with previous code this was more readable and clear

File: kafka/producer/src/main/java/ProducerConfiguration.java [hunk 8]
```diff
@@ -99,9 +99,9 @@ public static Properties createProperties(ProducerConfiguration config) {
             }
         }
 
-        if ((config.getOauthAccessToken() != null)
-            || (config.getOauthTokenEndpointUri() != null && config.getOauthClientId() != null && config.getOauthRefreshToken() != null)
-            || (config.getOauthTokenEndpointUri() != null && config.getOauthClientId() != null && config.getOauthClientSecret() != null))    {
+        if (config.getOauthAccessToken() != null
+            || config.getOauthTokenEndpointUri() != null && config.getOauthClientId() != null && config.getOauthRefreshToken() != null
+            || config.getOauthTokenEndpointUri() != null && config.getOauthClientId() != null && config.getOauthClientSecret() != null)    {
```
Comment (im-konge) [inline kafka/producer/src/main/java/ProducerConfiguration.java:104]: Same as above

File: kafka/streams/src/main/java/StreamsConfiguration.java [hunk 9]
```diff
@@ -96,8 +96,8 @@ public static Properties createProperties(StreamsConfiguration config) {
             }
         }
 
-        if ((config.getOauthAccessToken() != null)
-            || (config.getOauthTokenEndpointUri() != null && config.getOauthClientId() != null && (config.getOauthRefreshToken() != null || config.getOauthClientSecret() != null)))    {
+        if (config.getOauthAccessToken() != null
+            || config.getOauthTokenEndpointUri() != null && config.getOauthClientId() != null && (config.getOauthRefreshToken() != null || config.getOauthClientSecret() != null))    {
```
Comment (im-konge) [inline kafka/streams/src/main/java/StreamsConfiguration.java:100]: Same as above

**Review by im-konge — COMMENTED** [review]
Several nits, as everything was already mentioned. Also, how I suggested, would it make sense to have this tracing stuff in separate repo and then just use dependency of it in all repos like `operators`, `test-client` and with little bit of change `bridge`?

File: docker-images/build-images.sh (hunk 1 above)
Comment (alesj) [reply docker-images/build-images.sh:15]: All other modules depend on this one -- so it needs to be built before.

File: Makefile (hunk 2 above)
Comment (alesj) [reply Makefile:19]: OK, I'll move it.

File: http/http-producer/pom.xml (hunk 3 above)
Comment (alesj) [reply http/http-producer/pom.xml:40]: All tracing (and this includes OpenTracing) moved to `tracing` module.
And it's all back compatible.

File: kafka/admin/src/main/java/AdminConfiguration.java (hunk 4 above)
Comment (alesj) [reply kafka/admin/src/main/java/AdminConfiguration.java:98]: This was forced by checkstyle plugin.
We can either leave it as it was and ignore the checkstyle error, 
or apply the checkstyle rules, your call.

File: tracing/src/main/java/io/strimzi/test/tracing/TracingUtil.java (hunk 6 above)
Comment (alesj) [reply tracing/src/main/java/io/strimzi/test/tracing/TracingUtil.java:18]: OK, changing to log4j.

File: kafka/producer/src/main/java/Main.java (hunk 7 above)
Comment (alesj) [reply kafka/producer/src/main/java/Main.java:74]: Checkstyle plugin ...

**Comment by alesj** [comment]
> Also, how I suggested, would it make sense to have this tracing stuff in separate repo and then just use dependency of it in all repos like operators, test-client and with little bit of change bridge?

Yes, I agree.
But otoh we all know we don't just go out and create new repos in `strimzi` org for any small stuff.
@scholzj what do you suggest?

**Comment by alesj** [comment]
A short description of the PR ...

The new `tracing` module abstracts away the tracing usage -- either previous OpenTracing or new OpenTelemetry support. Instead of same tracing config code being applied over all test-client modules, we handle this in a single `tracing` module.
Different tracing implementations are triggered by setting implementation specific `service name` environment variable (see tracing's README for more details).
Each tracing implementation sets a global `Tracer` instance, which can then be used in supported libraries to provide new traces / spans.
Additionally to `service name` env var, other required env vars must be set as well -- see README.
Currently tracing implementations add support for Kafka consumer/producer and Kafka Streams tracing.

Similar implementation is used in Strimzi Operator and Strimzi Bridge,
so it might be useful to reconsider a better place for all this implementation code / artifact.

File: http/pom.xml [hunk 10]
```diff
@@ -41,19 +41,9 @@
...
-                <artifactId>opentracing-kafka-client</artifactId>
-                <version>${opentracing-kafka.version}</version>
-            </dependency>
-            <dependency>
-                <groupId>io.opentracing.contrib</groupId>
-                <artifactId>opentracing-kafka-streams</artifactId>
-                <version>${opentracing-kafka.version}</version>
+                <groupId>io.strimzi</groupId>
+                <artifactId>test-client-tracing</artifactId>
+                <version>${project.version}</version>
```
Comment (scholzj) [inline http/pom.xml:46]: Is the version needed?

File: kafka/admin/pom.xml [hunk 11]
```diff
@@ -28,14 +28,6 @@
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-simple</artifactId>
         </dependency>
-        <dependency>
```
Comment (scholzj) [inline kafka/admin/pom.xml:31]: Again, why are these somewhere replaced with `test-client-tracing` and somewhere just deleted?

File: kafka/admin/src/main/java/AdminConfiguration.java (hunk 4 above)
Comment (scholzj) [reply kafka/admin/src/main/java/AdminConfiguration.java:98]: There was no checkstyle error before => so why you have checkstyle error now?

File: kafka/consumer/src/main/java/Main.java [hunk 12]
```diff
@@ -3,22 +3,18 @@
  * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
  */
 
-import io.jaegertracing.Configuration;
-import io.opentracing.Tracer;
-import io.opentracing.contrib.kafka.TracingConsumerInterceptor;
-import io.opentracing.util.GlobalTracer;
-import org.apache.kafka.clients.consumer.ConsumerConfig;
+import java.time.Duration;
```
Comment (scholzj) [inline kafka/consumer/src/main/java/Main.java:6]: Keep the original order of imports.

File: kafka/producer/src/main/java/Main.java [hunk 13]
```diff
@@ -3,43 +3,34 @@
...
-import org.apache.logging.log4j.LogManager;
-
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Properties;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Future;
 import java.util.concurrent.atomic.AtomicLong;
 
+import io.strimzi.test.tracing.TracingUtil;
```
Comment (scholzj) [inline kafka/producer/src/main/java/Main.java:13]: Imports order.

File: pom.xml [hunk 14]
```diff
@@ -42,7 +48,7 @@
                     <dependency>
                         <groupId>com.puppycrawl.tools</groupId>
                         <artifactId>checkstyle</artifactId>
-                        <version>8.29</version>
+                        <version>9.0.1</version>
```
Comment (scholzj) [inline pom.xml:51]: Seems unrelated to this PR. Please do a separate Pr for it if you want to update it.

File: tracing/README.md [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+
+List of env vars we need to set:
+
+#### OpenTracing
+ * JAEGER_SERVICE_NAME -- this triggers OpenTracing tracing
+ * JAEGER_AGENT_HOST -- host name where the Jaeger traces are sent
+
+#### OpenTelemetry
+ * OTEL_SERVICE_NAME -- this triggers OpenTelemetry tracing
+ * OTEL_EXPORTER_JAEGER_ENDPOINT -- url where the Jaeger traces are sent
```
Comment (scholzj) [inline tracing/README.md:19]: Since you expect both to feed into Jaeger ... does the expected `JAEGER_AGENT_HOST` differ from `OTEL_EXPORTER_JAEGER_ENDPOINT`? Or is it expected to be the same?

File: tracing/README.md [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+ * OTEL_TRACES_EXPORTER=jaeger (this is not required, as it's done by the run.sh script)
```
Comment (scholzj) [inline tracing/README.md:20]: What are the other options? Or why do we need this option?

File: http/http-consumer/pom.xml [hunk 17]
```diff
@@ -31,14 +31,6 @@
             <artifactId>log4j-slf4j-impl</artifactId>
             <version>${log4j.version}</version>
         </dependency>
-        <dependency>
```
[the remark this answers is not in the document]
Comment (alesj) [reply http/http-consumer/pom.xml:34]: Http-consumer had no tracing code.
Http-producer had it ... although I doubt it was used / kicked-in ...
@im-konge ^ ?

[the remark this answers is not in the document]
Comment (im-konge) [reply http/http-consumer/pom.xml:34]: I suppose that was some mistake, so if you have it in http-producer, it should be in consumer as well. Can you please add it there?

File: http/http-producer/src/main/java/HttpProducerApp.java [hunk 18]
```diff
@@ -3,15 +3,13 @@
...
-import io.jaegertracing.Configuration;
-import io.opentracing.Tracer;
-import io.opentracing.util.GlobalTracer;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
-
 import java.net.URISyntaxException;
 import java.util.concurrent.ExecutionException;
 
+import io.strimzi.test.tracing.TracingUtil;
```
[the remark this answers is not in the document]
Comment (alesj) [reply http/http-producer/src/main/java/HttpProducerApp.java:9]: OK.

Where do static imports go?
After project's or at the bottom?

File: http/pom.xml (hunk 10 above)
Comment (alesj) [reply http/pom.xml:46]: This is how it was done before -- all deps in http/pom.xml have versions.

File: kafka/admin/pom.xml (hunk 11 above)
Comment (alesj) [reply kafka/admin/pom.xml:31]: Admin had no tracing code.

File: kafka/consumer/src/main/java/Main.java (hunk 12 above)
Comment (alesj) [reply kafka/consumer/src/main/java/Main.java:6]: > There was no checkstyle error before => so why you have checkstyle error now?

Checkctyle upgrade ...

File: pom.xml (hunk 14 above)
Comment (alesj) [reply pom.xml:51]: OK

File: tracing/README.md (hunk 15 above)
Comment (alesj) [reply tracing/README.md:19]: I think they differ.
At least in the Strimzi CO test they were different.

File: tracing/README.md (hunk 16 above)
Comment (alesj) [reply tracing/README.md:20]: Other options would be other supported tracing frameworks; e.g. `zipkin`
It needs to be explicitly set -- it doesn't know how to "read" this from the classpath ...

File: http/http-consumer/pom.xml (hunk 17 above)
[the remark this answers is not in the document]
Comment (scholzj) [reply http/http-consumer/pom.xml:34]: So if the consumer doesn't need it, it can be just removed as it was, or?

File: http/http-producer/src/main/java/HttpProducerApp.java (hunk 18 above)
[the remark this answers is not in the document]
Comment (scholzj) [reply http/http-producer/src/main/java/HttpProducerApp.java:6]: Normally static imports are at the end of all the imports IIRC.

File: tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java [hunk 19]
```diff
@@ -0,0 +1,97 @@
...
+    public String serviceName() {
+        String serviceName = System.getenv(envName());
+        if (serviceName == null) {
+            serviceName = System.getProperty("otel.service.name");
+        } else {
+            System.setProperty("otel.service.name", serviceName);
+        }
+        if (serviceName != null && System.getenv("OTEL_TRACES_EXPORTER") == null) {
+            System.setProperty("otel.traces.exporter", "jaeger");
+        }
```
Comment (scholzj) [inline tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java:45]: You set the default here as well as in the bash script ... would it be cleaner to do it only in one place?

File: docker-images/build-images.sh [hunk 20]
```diff
@@ -7,6 +7,12 @@ MODE=${1:-"push"}
 DOCKER_TAG=${DOCKER_TAG:-"latest"}
 MVN_ARGS=${MVN_ARGS:-""}
 
+echo "Installing root pom"
+make java_install_single
+
+echo "Building Tracing dependency"
+make build --directory=tracing/
```
[the remark this answers is not in the document]
Comment (alesj) [reply docker-images/build-images.sh:14]: Check whole `build-images.sh` ... as it was done before, it only picked-up `kafka` and `http` modules, since they only had lib dependencies.
Now there is `tracing` module dependency, which needs to be build before.
This was one way of achieving this ... I'm fine with any other way.

File: tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java (hunk 19 above)
Comment (alesj) [reply tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java:45]: This is just in case there is no run script.
I can remove it.

File: tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java [hunk 21]
```diff
@@ -0,0 +1,97 @@
...
+
+public class OpenTelemetryHandle implements TracingHandle {
+    @Override
+    public String type() {
+        return "OpenTelemetry";
+    }
+
+    @Override
+    public String envName() {
+        return "OTEL_SERVICE_NAME";
```
[the remark this answers is not in the document]
Comment (alesj) [reply tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java:32]: New compilers inline constants.

File: tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java (hunk 19 above)
Comment (alesj) [reply tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java:45]: > Well, It didn't had test-client-tracing dependency from the same project before. So I think you can remove it here.

It needs to be defined in one place.
It looked like this was the place before -- for any libs used in sub modules.

Comment (scholzj) [reply tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java:45]: > This is just in case there is no run script.
I can remove it.

Why don't you remove it from the script? I think is makes sense _ just in case there is no run script_ ... but why have it in the run script then?

File: tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java (hunk 21 above)
[the remark this answers is not in the document]
Comment (scholzj) [reply tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java:32]: I guess it would anyway make the code more readable to have the constants defined in one place, or?

File: docker-images/build-images.sh (hunk 1 above)
Comment (scholzj) [reply docker-images/build-images.sh]: Normally, the submodule would be added into the `SUBDIRS` list in the main Makefile. So when you run `make ...` it would trigger it for the tracing module as well and build it that way.

The README.md suggests to call `make all` to build the project. But if this is only in the `build-images.sh` script, it would be never called and I assume it would not work with it. That is at least my read on the build.

Comment (alesj) [reply docker-images/build-images.sh]: Atm the script iterates over `kafka` and `http` subdirs, and builds them.
Whereas `tracing` module is a seperate module on which they depend on,
so it must be built before.

This build-images.sh is only used in CI, afais.
The plain Make makes use of `SUBDIRS`, where `tracing` is listed as well.

If @im-konge or @Frawless have any better ideas, fine by me.

File: tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java (hunk 21 above)
[the remark this answers is not in the document]
Comment (alesj) [reply tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java:32]: That's the only place where tracing impl specific const are used.
Imo it makes it much clearer to be in there, where as in some global `Constants` class.

[the remark this answers is not in the document]
Comment (scholzj) [reply tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java:32]: I'm not talking about any Constants class ... I'm talking about having all of them defined as a constants (static final fields) in this class.

[the remark this answers is not in the document]
Comment (alesj) [reply tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java:32]: Why is that better?

[the remark this answers is not in the document]
Comment (alesj) [reply tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java:32]: They already are constants -- done by the compiler.

File: tracing/src/main/java/io/strimzi/test/tracing/TracingUtil.java [hunk 22]
```diff
@@ -0,0 +1,83 @@
...
+
+        @Override
+        public void kafkaProducerConfig(Properties props) {
+        }
+
+        @Override
+        public KafkaClientSupplier clientSupplier() {
+            return null;
+        }
+    }
```
Comment (im-konge) [inline tracing/src/main/java/io/strimzi/test/tracing/TracingUtil.java:81]: Would make sense to do this:
- change the `TracingHandle` to be `abstract class`
- have there these `null`s
- `kafkaConsumerConfig` & `kafkaProducerConfig` are same for both tracing options, so having them in abstract would save the overrides
- then instead of `private static TracingHandle tracing = new NoopTracing();` you would do just `    private static TracingHandle tracing = new TracingHandle()` and you don't have to do another class in class?

Comment (alesj) [reply tracing/src/main/java/io/strimzi/test/tracing/TracingUtil.java:81]: It's just another no-impl class, no real overhead ... and it clearly shows a no-op.
Actual impl have all methods different -- `kafkaConsumerConfig` & `kafkaProducerConfig` are different, just same class name, diff package.
I prefer interfaces over actual classes -- to point out the needed signature.
But if you insist on this, np ...

Comment (im-konge) [reply tracing/src/main/java/io/strimzi/test/tracing/TracingUtil.java:81]: It was just my point, if it makes more sense to have it like you have, then keep it as it is.

Comment (alesj) [reply tracing/src/main/java/io/strimzi/test/tracing/TracingUtil.java:81]: Yeah, not a bad point -- when things are duplicated, which is not the case here -- things looks similar (since they are all tracing, no magic there ...), but differ in impl details (or packages).
We could have a default impl in the interface (nulls, no-op, etc) or an abstract class with these defaults.
But it doesn't hurt to be more explicit with Noop impl.

**Comment by alesj** [comment]
Closing due to #16

File: kafka/admin/src/main/java/AdminConfiguration.java (hunk 4 above)
Comment (alesj) [reply kafka/admin/src/main/java/AdminConfiguration.java:98]: Braces are back + added suppress ...

File: http/http-consumer/pom.xml (hunk 17 above)
[the remark this answers is not in the document]
Comment (alesj) [reply http/http-consumer/pom.xml:34]: Both http modules now have tracing support.

File: http/pom.xml (hunk 10 above)
Comment (alesj) [reply http/pom.xml:46]: So it should go into each submodule?

File: tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java (hunk 19 above)
Comment (alesj) [reply tracing/src/main/java/io/strimzi/test/tracing/OpenTelemetryHandle.java:45]: It's removed from the script.

File: tracing/src/main/java/io/strimzi/test/tracing/TracingUtil.java (hunk 22 above)
Comment (alesj) [reply tracing/src/main/java/io/strimzi/test/tracing/TracingUtil.java:81]: I left the Noop ...
